### ISSUE TITLE
Port the application to mobile platforms

### ISSUE THREAD

**Issue by raxi2012** [xpipe-io/xpipe#99]
There are almost no universal applications for working with remote servers on iOS/Android, and those that do exist operate exclusively on a subscription basis, without the option of a one-time purchase, such as Termius.

**Comment by crschnick** [comment]
I can see some challenges for a mobile application implementation. Mainly the dependency on other command-line programs as xpipe doesn't ship with any libraries for ssh or others. And apps tend to be sandboxed, which would prevent calling any other apps/executables

The sandboxing is even a problem with some package managers like flatpak, it's impossible to create an xpipe flatpak as that can't interact with other commandline tools.

I also think that it's pretty difficult to create a mobile application if it was not originally envisioned as one. You would have to change so many things, at that point it is basically an unrelated application to the original. In theory, the UI framework JavaFX supports mobile applications though.

I can leave this issue open for the future, but don't expect things to happen very soon as that will take a lot of work. First I have to evaluate how restrictive the sandboxing actually is and whether there are solutions to it.

**Comment by tcatlas** [comment]
I'll just note that with iOS, it is against the App Store guidelines for apps to retrieve code on-demand, so any loading of libraries must be done through them being included in the app package.

That said, I am very interested in any competitive, paid or no, alternative to Termius as they are overpriced and do not take customer feedback seriously at all, and other apps such as Royal TS or Devolutions are too bulky.

**Comment by crschnick** [comment]
I'm aware of the many mobile limitations, that is why there were no updates to this issue. As of right now I don't have the capabilities to implement a mobile build. I also have never done mobile development. Maybe in the future if operations are expanded there might be resources for that.

**Comment by tcatlas** [comment]
Yeah, honestly I think their prices should be cut in half. $5 for pro, $10 for team. Or $10 for pro and $10 for team under a certain number. Or a package like 1pass has for small teams. $20/user/mo is hard to justify, even in a several hundred million dollar company, especially when I can get the entire Microsoft 365 suite with Business Premium including Intune, Defender, Entra P1 and more for $18.50/user/mo…

**Comment by crschnick** [comment]
When people talk about mobile support, do you really mean like smartphones or more like various larger tablets? Maybe that's just me, but I couldn't imagine using any kind of SSH client on a small smartphone screen with a touchscreen.

**Comment by crschnick** [comment]
I see. The issue is that a mobile lap is a big challenge to implement. The only thing that I can remotely offer to mobile users is to try out https://github.com/xpipe-io/xpipe-webtop from the browser. But that requires setup on your own servers and a proper authentication setup to reach it remotely from where you host it.

**Comment by nalladev** [comment]
@interbiznw
> I usually use it when not by a pc but needing to connect for some quick task(kill/restart process, restart container, etc) is necessary.

I second that.

**Comment by Taytay** [comment]
Came here to say that I am using Termius quite a bit to log in and interact with Claude Code on a remote server. That increases the usability of an ssh session on such a small screen. Termius gives me a fraction of the features of XPipe though, so having all of my machines in one place on mobile would still be helpful.

**Comment by crschnick** [comment]
I recently stumbled upon the new Android Linux Terminal, which is essentially just a graphical Linux VM running on Android using the new AVF framework. If you are running Android 16 and your phone supports AVF (not all do, even newer ones), you can use it: https://www.zdnet.com/article/how-to-use-the-new-linux-terminal-on-android/

It actually works decently well with some adjustments:

<img width="4032" height="3024" alt="Image" src="https://github.com/user-attachments/assets/f721a6c8-7d68-4023-9c20-6ad49c38b0fd" />

The fixes for this platform are available in https://github.com/xpipe-io/xpipe-ptb/releases/tag/20.0-10 if anyone is interested. You can just install it as a normal .deb installer on the system and it should work automatically. The installer script `bash <(curl -sL https://github.com/xpipe-io/xpipe/raw/master/get-xpipe.sh) -s -v 20.0-10` is the easiest option.

Let me know what you think about this and whether it works for you. The whole AVF system is still very new and a bit unstable, but I hope that over time this becomes a viable solution for anyone on an Android device.

**Comment by AdamsGH** [comment]
> I recently stumbled upon the new Android Linux Terminal, which is essentially just a graphical Linux VM running on Android using the new AVF framework. If you are running Android 16 and your phone supports AVF (not all do, even newer ones), you can use it: https://www.zdnet.com/article/how-to-use-the-new-linux-terminal-on-android/
> 
> It actually works decently well with some adjustments:
> 
> <img width="4032" height="3024" alt="Image" src="https://github.com/user-attachments/assets/f721a6c8-7d68-4023-9c20-6ad49c38b0fd" />
> 
> The fixes for this platform are available in https://github.com/xpipe-io/xpipe-ptb/releases/tag/20.0-10 if anyone is interested. You can just install it as a normal .deb installer on the system and it should work automatically. The installer script `bash <(curl -sL https://github.com/xpipe-io/xpipe/raw/master/get-xpipe.sh) -s -v 20.0-10` is the easiest option.
> 
> Let me know what you think about this and whether it works for you. The whole AVF system is still very new and a bit unstable, but I hope that over time this becomes a viable solution for anyone on an Android device.

AVF is really wonderful, but while you have Snapdragon Elite - it's mostly locked for you. So came here just to leave one more «pleeeease» for native android app)
